### ISSUE TITLE
Compilation errors on Ubuntu 22.04 LTS with GCC 10 and Boost

### ISSUE THREAD

**Issue by ghost** [BeamMP/BeamMP-Server#177]
OS: Ubuntu 22.04 LTS
BeamMP-Server Version: tags/v3.1.1

Describe the bug
During the build process of BeamMP-Server, the following errors were encountered:

- error: ‘gettid’ was not declared in this scope; did you mean ‘getgid’?
- error: class ‘Version’ does not have any field named ‘gnu_dev_major’
- error: class ‘Version’ does not have any field named ‘gnu_dev_minor’
- g++-10: fatal error: Killed signal terminated program cc1plus

**To Reproduce**

1. Clone the BeamMP-Server repository
2. Switch to the latest release tag (v3.1.1)
3. Build BeamMP-Server using Boost 1.80.0 and CMake 3.26.1.

**Expected behavior**

The build process should complete without any errors.

**Additional context**

The build was attempted on a clean installation of Ubuntu 22.04 LTS, and Boost was installed using the system package manager. Any assistance in resolving this issue would be greatly appreciated.

**Comment by lionkor** [comment]
Please provide the full cmake + compiler output, in raw text. I removed the BUG label for you, as this isn't a bug as far as I can see :)

**Comment by lionkor** [comment]
Please also remember the [Contributing.md](https://github.com/BeamMP/BeamMP-Server/blob/master/CONTRIBUTING.md) and thus make sure to understand which branch to base your work off of.

**Comment by ghost** [comment]
> Please provide the full cmake + compiler output, in raw text. I removed the BUG label for you, as this isn't a bug as far as I can see :)

Ran it a second time just to make sure this is persistent. Here is my output:

```
azureuser@BeamMP-001:~/BeamMP-Server/build-linux$ sudo cmake ..
-- You can find build instructions and a list of dependencies in the README at https://github.com/BeamMP/BeamMP-Server
-- Submodule update
-- Checking for Sentry URL
-- SENTRY_TRANSPORT=curl
-- SENTRY_BACKEND=breakpad
-- SENTRY_LIBRARY_TYPE=STATIC
-- SENTRY_SDK_NAME=
-- Adding local source dependencies
-- Module support is disabled.
-- Version: 9.0.0
-- Build type: Release
-- CXX_STANDARD: 17
-- Required features: cxx_variadic_templates
-- Configuring done (1.0s)
-- Generating done (0.2s)
-- Build files have been written to: /home/azureuser/BeamMP-Server/build-linux
azureuser@BeamMP-001:~/BeamMP-Server/build-linux$ sudo make install
[  2%] Built target fmt
[ 24%] Built target breakpad_client
[ 57%] Built target sentry
[ 59%] Built target commandline
[ 60%] Building CXX object CMakeFiles/BeamMP-Server.dir/src/Common.cpp.o
/home/azureuser/BeamMP-Server/src/Common.cpp:294:13: warning: In the GNU C Library, "major" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "major", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "major", you should undefine it after including <sys/types.h>.
  294 |     : major(major)
      |             ^~~~~~                                                                                                                                                                                                                                                                                                                                                                  
/home/azureuser/BeamMP-Server/src/Common.cpp:295:13: warning: In the GNU C Library, "minor" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "minor", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "minor", you should undefine it after including <sys/types.h>.
  295 |     , minor(minor)
      |             ^~~~~~                                                                                                                                                                                                                                                                                                                                                                  
/home/azureuser/BeamMP-Server/src/Common.cpp: In function ‘void RegisterThread(const string&)’:
/home/azureuser/BeamMP-Server/src/Common.cpp:278:31: error: ‘gettid’ was not declared in this scope; did you mean ‘getgid’?
  278 |     ThreadId = std::to_string(gettid());
      |                               ^~~~~~
      |                               getgid
In file included from /usr/include/x86_64-linux-gnu/sys/types.h:205,
                 from /usr/include/stdlib.h:394,
                 from /usr/include/c++/10/cstdlib:75,
                 from /usr/include/c++/10/ext/string_conversions.h:41,
                 from /usr/include/c++/10/bits/basic_string.h:6545,
                 from /usr/include/c++/10/string:55,
                 from /usr/include/c++/10/stdexcept:39,
                 from /usr/include/c++/10/system_error:41,
                 from /usr/include/c++/10/mutex:42,
                 from /home/azureuser/BeamMP-Server/include/TSentry.h:4,
                 from /home/azureuser/BeamMP-Server/include/Common.h:3,
                 from /home/azureuser/BeamMP-Server/src/Common.cpp:1:
/home/azureuser/BeamMP-Server/src/Common.cpp: In constructor ‘Version::Version(uint8_t, uint8_t, uint8_t)’:
/home/azureuser/BeamMP-Server/src/Common.cpp:294:7: error: class ‘Version’ does not have any field named ‘gnu_dev_major’
  294 |     : major(major)
      |       ^~~~~
/home/azureuser/BeamMP-Server/src/Common.cpp:295:7: error: class ‘Version’ does not have any field named ‘gnu_dev_minor’
  295 |     , minor(minor)
      |       ^~~~~
CMakeFiles/BeamMP-Server.dir/build.make:131: recipe for target 'CMakeFiles/BeamMP-Server.dir/src/Common.cpp.o' failed
make[2]: *** [CMakeFiles/BeamMP-Server.dir/src/Common.cpp.o] Error 1
CMakeFiles/Makefile2:255: recipe for target 'CMakeFiles/BeamMP-Server.dir/all' failed
make[1]: *** [CMakeFiles/BeamMP-Server.dir/all] Error 2
Makefile:135: recipe for target 'all' failed
make: *** [all] Error 2
azureuser@BeamMP-001:~/BeamMP-Server/build-linux$ 
```

**Comment by ghost** [comment]
For the version fields I decided to stick with the following as a temporary fix:

```h
#undef major
#undef minor
```

**Comment by ghost** [comment]
Was able to dumb it down to just this error after some minor adjustments to the code:

```
azureuser@BeamMP-001:~/BeamMP-Server/build-linux$ sudo make install
[  2%] Built target fmt
[ 24%] Built target breakpad_client
[ 57%] Built target sentry
[ 59%] Built target commandline
[ 60%] Building CXX object CMakeFiles/BeamMP-Server.dir/src/Common.cpp.o
/home/azureuser/BeamMP-Server/src/Common.cpp: In function ‘void RegisterThread(const string&)’:
/home/azureuser/BeamMP-Server/src/Common.cpp:278:31: error: ‘gettid’ was not declared in this scope; did you mean ‘getgid’?
  278 |     ThreadId = std::to_string(gettid());
      |                               ^~~~~~
      |                               getgid
CMakeFiles/BeamMP-Server.dir/build.make:131: recipe for target 'CMakeFiles/BeamMP-Server.dir/src/Common.cpp.o' failed
make[2]: *** [CMakeFiles/BeamMP-Server.dir/src/Common.cpp.o] Error 1
CMakeFiles/Makefile2:255: recipe for target 'CMakeFiles/BeamMP-Server.dir/all' failed
make[1]: *** [CMakeFiles/BeamMP-Server.dir/all] Error 2
Makefile:135: recipe for target 'all' failed
make: *** [all] Error 2
azureuser@BeamMP-001:~/BeamMP-Server/build-linux$ 
```

**Comment by ghost** [comment]
I'll place a `#include <sys/syscall.h>` and use the new version of gettid (SYS_gettid) and see if that solves the problem.

**Comment by ghost** [comment]
> I'll place a `#include <sys/syscall.h>` and use the new version of gettid (SYS_gettid) and see if that solves the problem.

Looks like it passed! 👍🏻 
Currently installing:

```
azureuser@BeamMP-001:~/BeamMP-Server/build-linux$ sudo make install
[  2%] Built target fmt
[ 24%] Built target breakpad_client
[ 57%] Built target sentry
[ 59%] Built target commandline
[ 60%] Building CXX object CMakeFiles/BeamMP-Server.dir/src/Common.cpp.o
[ 61%] Building CXX object CMakeFiles/BeamMP-Server.dir/src/Client.cpp.o
[ 62%] Building CXX object CMakeFiles/BeamMP-Server.dir/src/VehicleData.cpp.o
[ 63%] Building CXX object CMakeFiles/BeamMP-Server.dir/src/TConfig.cpp.o

```

**Comment by ghost** [comment]
Problem solved, thank you for your help and I hope this helps you find more things to improve! :)

**Comment by lionkor** [comment]
Pretty sure those are not cross platform compatible, so make sure your changes are either POSIX and behind macro guards (ifdefs) or work on all platforms.
